main.go

```diff
@@ -29,139 +30,221 @@ type backup struct {
 
-	log.Println("Evaluating backups")
+	// schedule evaluation of every backupConfig
+	for _, backupConfig := range backupConfigs {
+		s := gocron.NewScheduler(time.UTC)
+		job, err := s.Cron(evalSchedule).Do(evaluateBackups, dynamicClient, gvr, backupCount, oldestBackup, youngestBackup, backupConfig)
+		if err != nil {
+			log.Fatalf("%v: error creating job: %v", backupConfig.Name, err)
+		}
+		s.StartAsync()
+		log.Printf("%v: next run: %v\n", backupConfig.Name, job.NextRun())
+	}
+}
+
+func evaluateBackups(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, backupCount *prometheus.GaugeVec, oldestBackup *prometheus.GaugeVec, youngestBackup *prometheus.GaugeVec, backupConfig backupconfig) {
+
+	log.Printf("%v: evaluating backups\n", backupConfig.Name)
 
-	backups := getBackupActionsets(dynamicClient, gvr, kanisterNamespace)
+	backups := getBackups(dynamicClient, gvr, backupConfig)
 
-	dailyBackups, weeklyBackups := categoriseBackups(backups, dailyBackupsRetained, weeklyBackupsRetained)
+	categorisedBackups, failedBackupCount := categoriseBackups(backups, backupConfig)
 
-	dailyBackupCount.Set(float64(len(dailyBackups)))
-	weeklyBackupCount.Set(float64(len(weeklyBackups)))
+	maxBackups, err := strconv.Atoi(backupConfig.Retention.Backups)
+	if err != nil {
+		log.Printf("%v: error converting maxBackups to int: %v\n", backupConfig.Name, err)
+		os.Exit(1)
+	}
 
-	//if there are excess daily backups, delete the oldest excess
-	if len(dailyBackups) > dailyBackupsRetained {
-		deleteOldestBackups(dailyBackups, (len(dailyBackups) - dailyBackupsRetained), dynamicClient, gvr, kanisterNamespace, blueprintName, s3ProfileName)
+	// if there are excess daily backups, delete the oldest excess, then refetch and recategorise the backups
+	if len(categorisedBackups) > maxBackups {
+		deleteOldestBackups(categorisedBackups, (len(categorisedBackups) - maxBackups), dynamicClient, gvr, backupConfig)
+		backups = getBackups(dynamicClient, gvr, backupConfig)
+		categorisedBackups, failedBackupCount = categoriseBackups(backups, backupConfig)
 	} else {
-		log.Printf("No daily backups deleted: Current: %v Limit: %v\n", len(dailyBackups), dailyBackupsRetained)
+		log.Printf("%v: no backups deleted: current: %v limit: %v\n", backupConfig.Name, len(categorisedBackups), maxBackups)
 	}
 
-	//if there are excess weekly backups, delete the oldest excess
-	if len(weeklyBackups) > weeklyBackupsRetained {
-		deleteOldestBackups(weeklyBackups, (len(weeklyBackups) - weeklyBackupsRetained), dynamicClient, gvr, kanisterNamespace, blueprintName, s3ProfileName)
+	setPrometheusMetrics(categorisedBackups, backupCount, oldestBackup, youngestBackup, backupConfig, failedBackupCount)
+
+	log.Printf("%v: backup evaluation complete\n", backupConfig.Name)
+}
+
+func setPrometheusMetrics(backups []backup, backupCount *prometheus.GaugeVec, oldestBackup *prometheus.GaugeVec, youngestBackup *prometheus.GaugeVec, backupConfig backupconfig, failedBackupCount int) {
```

Comment:
now, this func could instead be a method (https://go.dev/tour/methods/1) of the struct I have mentioned in the above comment, its name could be something like `updateMetrics` and it would only take the data as argument (as the Prometheus vectors would already be part of the struct).

Reply:
done!